model-optimizer/unit_tests/extensions/ops/gather_test.py

```diff
@@ -3,59 +3,174 @@
-                           })
-
-    def test_gather_infer(self):
-        graph = self._create_graph()
-
-        gather_node = Node(graph, 'gather_node')
-        Gather.infer(gather_node)
-
-        exp_shape = int64_array([2, 15])
-        res_shape = graph.node['gather_output']['shape']
-        res_value = graph.node['gather_output']['value']
-
-        self.assertTrue(np.array_equal(exp_shape, res_shape),
-                        'shapes do not match expected: {} and given: {}'.format(exp_shape, res_shape))
-
-        self.assertTrue(np.array_equal(res_value, np.ones(exp_shape)),
-                        'shapes do not match expected: {} and given: {}'.format(exp_shape, res_shape))
+
+        edges = [
+            *connect('data', '0:gather'),
+            *connect('indices', '1:gather'),
+            *connect('axis', '2:gather'),
+            *connect('gather', 'res')
+        ]
+
+        graph = build_graph(nodes, edges)
+        graph.stage = 'middle'
+        partial_infer(graph)
+
+        node = Node(graph, 'gather')
+        res = node.out_port(0).data.get_shape()
+        npt.assert_array_equal(res, ref_shape)
+
+    def test_shape_axis_1(self):
+        self.build_and_test_shape_inference(axis=1, batch_dims=0,
+                                            data_shape=[3, 3],
+                                            indices_shape=[1, 2],
+                                            ref_shape=[3, 1, 2])
+
+    def test_shape_axis_0(self):
+        self.build_and_test_shape_inference(axis=0, batch_dims=0,
+                                            data_shape=[3, 3],
+                                            indices_shape=[1, 2],
+                                            ref_shape=[1, 2, 3])
+
+    def test_shape_axis_minus_2(self):
+        self.build_and_test_shape_inference(axis=-2, batch_dims=0,
+                                            data_shape=[2, 3, 7],
+                                            indices_shape=[1, 4],
+                                            ref_shape=[2, 1, 4, 7])
+
+    def test_shape_axis_1_batch_dims_1(self):
+        self.build_and_test_shape_inference(axis=1, batch_dims=1,
+                                            data_shape=[3, 4],
+                                            indices_shape=[3, 1, 2],
+                                            ref_shape=[3, 1, 2])
+
+    def test_shape_axis_2_batch_dims_1(self):
+        self.build_and_test_shape_inference(axis=2, batch_dims=1,
+                                            data_shape=[3, 4, 7],
+                                            indices_shape=[3, 1, 2],
+                                            ref_shape=[3, 4, 1, 2])
+
+    def test_shape_axis_2_batch_dims_minus_1(self):
+        self.build_and_test_shape_inference(axis=2, batch_dims=-1,
+                                            data_shape=[3, 4, 7],
+                                            indices_shape=[3, 1, 2],
```

Comment:
great catch! I will add additional check to shape_inference

Reply:
fixed this test, also added additional checks and negative tests. 
Since in checks I deal with original raw `axis` and `batch_dims` values and only them normalize them removed getters for `axis` and `batch_dims`. Please review